tests/unit/utility.test.ts

```diff
@@ -0,0 +1,145 @@
+      const range = createRangeFromLength(length);
+      expect(range).toStrictEqual([0, 1, 2, 3, 4]);
+    });
+  });
+
+  describe('isUndefined', () => {
+    it('correctly identifies if a value is undefined', () => {
+      const value = undefined;
+      const isValueUndefined = isUndefined(value);
+      expect(isValueUndefined).toBe(true);
+    });
+
+    it('correctly identifies if a value is not undefined', () => {
+      const value = 1;
+      const isValueUndefined = isUndefined(value);
+      expect(isValueUndefined).toBe(false);
+    });
+  });
+
+  describe('isDefined', () => {
+    it('correctly identifies if a value is defined', () => {
+      const value = 1;
+      const isValueDefined = isDefined(value);
+      expect(isValueDefined).toBe(true);
+    });
+
+    it('correctly identifies if a value is not defined', () => {
+      const value = undefined;
+      const isValueDefined = isDefined(value);
+      expect(isValueDefined).toBe(false);
+    });
+  });
+
+  describe('compareUint8Arrays', () => {
+    it('correctly compares two Uint8Arrays for equality', () => {
+      const uint8ArrayA = new Uint8Array([
+        0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x57, 0x6f, 0x72, 0x6c, 0x64, 0x21, 0xff, 0x00, 0x7f,
+        0x80, 0xfe, 0x01, 0x76, 0x31, 0x30,
+      ]);
+      const uint8ArrayB = new Uint8Array([
+        0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x57, 0x6f, 0x72, 0x6c, 0x64, 0x21, 0xff, 0x00, 0x7f,
+        0x80, 0xfe, 0x01, 0x76, 0x31, 0x30,
+      ]);
+      const areArraysEqual = compareUint8Arrays(uint8ArrayA, uint8ArrayB);
+      expect(areArraysEqual).toBe(true);
+    });
+
+    it('correctly identifies two different Uint8Arrays', () => {
+      const uint8ArrayA = new Uint8Array([
+        0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x57, 0x6f, 0x72, 0x6c, 0x64, 0x21, 0xff, 0x00, 0x7f,
+        0x80, 0xfe, 0x01, 0x76, 0x31, 0x30,
+      ]);
+      const uint8ArrayB = new Uint8Array([
+        0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x57, 0x6f, 0x72, 0x6c, 0x64, 0x21, 0xff, 0x00, 0x7f,
+        0x80, 0xfe, 0x01, 0x76, 0x31, 0x31,
+      ]);
+
+      const areArraysEqual = compareUint8Arrays(uint8ArrayA, uint8ArrayB);
+      expect(areArraysEqual).toBe(false);
+    });
+  });
+
+  describe('reverseBytes', () => {
+    it('correctly reverses the bytes of a Uint8Array', () => {
+      const uint8Array = new Uint8Array([0x48, 0x65, 0x6c, 0x6c, 0x6f]);
+      const reversedUint8Array = reverseBytes(uint8Array);
+      expect(reversedUint8Array).toStrictEqual(new Uint8Array([0x6f, 0x6c, 0x6c, 0x65, 0x48]));
+    });
+
+    it('correctly reverses the bytes of a Buffer', () => {
+      const buffer = Buffer.from([0x48, 0x65, 0x6c, 0x6c, 0x6f]);
+      const reversedBuffer = reverseBytes(buffer);
+      expect(reversedBuffer).toStrictEqual(Buffer.from([0x6f, 0x6c, 0x6c, 0x65, 0x48]));
+    });
+  });
+
+  describe('delay', () => {
+    it('delays execution for the specified number of milliseconds', () => {
+      const delayTime = 1000;
```

Comment:
lower this number so it doesn't slow down the test. 10?

Reply:
Alright!